src/label_nodes/CCLabelBMFont.cs

```diff
@@ -427,485 +436,417 @@ private int KerningAmountForFirst(int first, int second)
+        /// NOTE: rewritten 3/15/2015 (TOP 10)
+        ///     - #1 optomized so that there aren't 4+ interations through the string
+        ///     - #2 optomized to remove O(n) lookups for every character (4+ times) - e.g. CCBMFontConfiguration.CharacterSet (List<>)
+        ///     - #3 bug fix to remove crashing bugs because of trying to use this[i] == null
+        ///     - #4 bug fix to properly align horizontally -- wasn't calculating real character position
+        ///     - #5 rewrote in a more readable way, forever removed skip, previousSkip, etc.
+        ///     - #6 bug fix to properly handle whitespace at beginning and end of lines when doing horizontal alignment
+        ///     - #7 returning the correct ContentSize once calculated
+        ///     - #8 removed the second copy of labelText so there is now only 1
+        ///     - #9 now work with (and support) AnchorPoint
+        ///     - #10 now exposing ContentSize and LineHeight as return values after everything is calculated
+        ///     
+        ///     - and many more things which I have forgotten
+        /// 
+        /// NOTE2: n-spaces at the end of the line are basically ignored/removed if they take offset past the end of a line that needs to be split
+        /// 
+        /// NOTE3: There are some special cases which aren't handled consistently like what happens if the string starts with a bunch of blank spaces?
+        ///        It will different than if there are a string of blank spaces at the end of a line that is automatically broken.  Maybe this is as expected?
+        /// 
+        /// QUESTION: Would it make sense to Trim() the label before using or do you want to be able to add blank lines at the end?
+        ///           For now you can add blank lines anywhere.
+        /// </summary>
         public void CreateFontChars()
         {
-
-            int nextFontPositionX = 0;
-            int nextFontPositionY = 0;
-            char prev = (char) 255;
-            int kerningAmount = 0;
-
-            CCSize tmpSize = CCSize.Zero;
-
-            int longestLine = 0;
-            int totalHeight = 0;
-
-            int quantityOfLines = 1;
-
-            if (String.IsNullOrEmpty(labelText))
+            // Clear out all the sprites if (for example) we are changing the string
+            if (RecreateSprites)
             {
-                return; 
+                RemoveAllChildren(true);
+                RecreateSprites = false;
             }
 
-            int stringLen = labelText.Length;
+            // Values that we will calculate
+            base.ContentSize = CCSize.Zero;
 
-            var charSet = FontConfiguration.CharacterSet;
-            if (charSet.Count == 0)
+            // Quit if there's nothing to build
+            if (string.IsNullOrEmpty(labelText))
             {
-                throw (new InvalidOperationException(
-                    "Can not compute the size of the font because the character set is empty."));
```

Comment:
Yes, and feel free to remove all of my comments referring to legacy stuff -- I added them for your review and in case I made a mistake because I didn't understand some other dependency.